src/main/java/com/google/sps/servlets/YoutubePost.java

```diff
@@ -14,20 +14,58 @@
 
 package com.google.sps.servlets;
 
+import com.google.api.client.util.DateTime;
+import com.google.api.services.youtube.YouTube.CommentThreads;
+import com.google.api.services.youtube.model.CommentListResponse;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.api.services.youtube.model.Video;
+import com.google.api.services.youtube.model.VideoListResponse;
+import com.google.sps.servlets.YoutubeApi;
 import com.google.sps.servlets.YoutubeApiException;
+import java.math.BigInteger;
+import java.util.List;
 
 /** Data model for a singular Youtube post that will be added into a list of posts. */
 public class YoutubePost {
   private final String title;
   private final String content;
   private final String url;
   private final String id;
+  private static DateTime timeStamp;
+  private final List<String> comments;
+  private final List<Long> commentLikesList;
+  private static BigInteger likeCount;
 
-  public YoutubePost(String title, String content, String id) {
+  public YoutubePost(String title, String content, String id, List<String> comments,
+      List<Long> commentLikesList, DateTime timeStamp) {
     this.title = title;
     this.content = content;
     this.id = id;
     this.url = "https://www.youtube.com/watch?v=" + id;
+    this.comments = comments;
+    this.commentLikesList = commentLikesList;
```

Comment:
it's always preferable to copy the input list and make it immutable if possible:

```java
this.comments = ImmutableList.copyOf(comments);
```

This is because if an external object is modifying the array, it would result in the internal array being updated as well (because they're the same object). With a copy, we don't have to worry about that because we know the array won't change.